Library/Formula/qt5.rb

```diff
@@ -72,6 +74,13 @@ class Qt5 < Formula
   depends_on OracleHomeVarRequirement if build.with? "oci"
 
   def install
+    # Save users some frustration by telling them that `--cc=gcc-5` or similar
+    # won't work. It won't be picked up by the Qt build system, there is no easy
+    # way to select an arbitrary compiler, and it's not supported anyway.
+    if ENV.compiler != :clang
```

Comment:
My last comment sounds more rude than I was hoping it would. Sorry!

I'd be alright without this part and just raising the minimum OS X requirement (the primary goal of this PR). I could revise this part once a better solution in core is available (either implemented by me or someone else). How does that sound?